### ISSUE TITLE
Fuskr doesn't show any context menu

### ISSUE THREAD

**Issue by jedikalimero** [DanAtkinson/Fuskr#44]
Using Firefox 71.0. Tried to disable all other extensions just in case there was some interference but Fuskr doesn't work. Right click on images that contain numbers does not show any Fusk menu.

All comments at https://addons.mozilla.org/firefox/addon/fuskr/reviews/ indicate the same issue.

**Comment by jedikalimero** [comment]
It seems in Firefox 72.0.1 Fuskr is working correctly again.

**Comment by jedikalimero** [comment]
Ups! No, it is not working again!
Still in Firefox 72.0.1, but Fuskr has disappeared again from context menu or toolbar.
I have disabled all add ons except Fuskr and restarted Firefox but it still doesn't show up when it should.

**Comment by jedikalimero** [comment]
I have tried several old versions of Fusker in Firefox Quantum 64-bit on Windows 10 64-bit but none of them seems to work.
In VERY VERY RARE ocassions, Fusker has appeared in the context menu and worked as intended. I have tried to reproduce the situation but it has been not possible to determine when it works

**Comment by sloanlance** [comment]
I suspect Fuskr is having conflicts with other Chrome extensions I have installed, but I've not yet taken time to set up a minimal profile for an experiment.  (Or bothered to disable all other extensions.)

**Comment by jedikalimero** [comment]
I have disabled ALL firefox extensions and restarted Firefox and Fusker still doesn't work.

**Comment by sloanlance** [comment]
I guess there's always JavaScript fuskers, like http://members.home.nl/bdr/files/fusker.html.

**Comment by jedikalimero** [comment]
> I guess there's always JavaScript fuskers, like http://members.home.nl/bdr/files/fusker.html.

Of course, but the point of using an add-on is to right click on the image and select a menu option that does it all for you.

With that Javascript you need to copy the url of the image, open the url of the script, paste the clipboard and hit the button. Also, Fuskr is supposed to be a little moro complex than that, allowing to download the images, define the range, edit the regular expression in case the index is not the rightmost number found in the url, managing missing images in the sequence and leading zeros, etc.

> I wonder whether it's non-functioning status has anything to do with:
> 
> * Searching for "fuskr" or "fusker" in the Chrome store turns up nothing.  ("fusk" brings up a couple of games.)
> * The Chrome store URL for this extension no longer works (https://chrome.google.com/webstore/detail/fuskr/glieaboaghdnlglpkekghloldikefofo).

It is available in the Firefox add-ons store but I wonder if the author has any intention on updating Fuskr. It has been a long time since the last version.
I'm not sure how all this thing of Github goes but I believe all the code is available for anybody who wants to go on where the author left it (I think it is called a "fork"). Unfortunately, I'm just a graphic designer so I have no idea of Javascript or any other programming language.

**Comment by DanAtkinson** [comment]
Hi,

I have no plans to release an update at this time. Unfortunately, all of my free time is being taken up with my professional and personal life leaving me with very little time for Fuskr.

The extension is no longer in the Chrome Web Store either. A few months ago (20th February), I got an email from Google stating that they intended to remove my extension from the store because it did not meet the "User data privacy".

For reference, here is what they sent me:

> Your product violates the 'Use of Permissions' section of the policy, which requires that you:
> *   Request access to the narrowest permissions necessary to implement your product’s features or services.
> *   If more than one permission could be used to implement a feature, you must request those with the least access to data or functionality.
> *   Don't attempt to 'future proof' your product by requesting a permission that might benefit services or features that have not yet been implemented.

For what it's worth, the extension permissions haven't changed and all permissions were in use. Here are all of the permissions from the manifest.json with a rundown of what they're used for:

* [**`activeTab`**](https://developer.chrome.com/extensions/activeTab)
  The `activeTab` permission allows extensions to interrogate the current url when the extension is invoked. This allowed the application to read the active tab url (in order to generate a fusk gallery), but again, this only happens when the extension is clicked, say in the context menu...
* [**`contextMenus`**](https://developer.chrome.com/extensions/contextMenus)
  The `contextMenu` permission rather predictably allows you to create a context menu. However, this particular permission does appear to be broken or at least different in Firefox and is the crux of this issue.
* [**`downloads`**](https://developer.chrome.com/extensions/downloads)
  The `download` permissions allows the extension to initiate downloads. This is used for generating and pushing the downloading of zip files to the user.
* [**`storage`**](https://developer.chrome.com/extensions/storage)
  The `storage` permission allows the extension to store information. Specifically it stored your last 10 fusks and your options. Nothing else.

As shown, all permissions are required by the application and none of them are for future proofing. I don't store any extension data and the source code for the extension is the same as the source code in this repository.

There was no way to flag this as an error with Google so I decided to pre-empt any forced removal by simply removing the extension myself. I'd become increasingly frustrated by them in recent months so this wasn't a problem at all for me.

I stopped using Google Chrome over a year ago and now only use it for testing purposes. My primary browser is Firefox but interestingly (and perhaps frustratingly) I do not have the same issues regarding the context menu going missing in it. The extension works perfectly for me and, if it wasn't working, then I'd almost certainly spend some time looking at and fixing it. However, because it's fine, I haven't, and so the problem appears to be there for others as well.

I will get around to looking at it _at some point_ but I'm afraid that time is not yet.

If, in the meantime you know how to code and want to play around, I actively encourage you download the source, make some changes, and commit a pull request if you fix the problem.

I apologise if this isn't the answer you wanted.

I will leave this issue open for further discourse and (hopefully) a potential fix.

Best wishes, Dan
